packages/vite-plugin-checker/src/checkers/typescript/main.ts

```diff
@@ -109,11 +109,31 @@ const createDiagnostic: CreateDiagnostic<'typescript'> = (pluginConfig) => {
       // https://github.com/microsoft/TypeScript/issues/32385
       // https://github.com/microsoft/TypeScript/pull/33082/files
       const createProgram = ts.createEmitAndSemanticDiagnosticsBuilderProgram
+      const addtionalTypescriptOptions =
+        typeof pluginConfig.typescript === 'object' ? pluginConfig.typescript.extraTSOptions : {}
 
       if (typeof pluginConfig.typescript === 'object' && pluginConfig.typescript.buildMode) {
         const host = ts.createSolutionBuilderWithWatchHost(
           ts.sys,
-          createProgram,
+          /* eslint-disable */
```

Comment:
**question:** Why disable eslint @EliLichtblau?

Reply:
Theres a lint rule that makes it impossible to make a function that takes I think more than 3 arguments iirc - this needs to take 6

Reply:
Try using [Rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) instead. Disabling eslint should be the last resort i guess.

Reply:
Maybe philosophical difference but I disable eslint when making eslint happy makes the code harder to understand. Imo rest parameters make this code that is currently clear less clear. If the maintainer asks me I'll happily make the change but until then I think disabling eslint is probably the right call